pkg/vault/create.go

```diff
@@ -0,0 +1,27 @@
+package vault
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/vault/api"
+	"go.opencensus.io/trace"
+)
+
+func (vc *vaultClient) Create(engine, key, value string) (*api.Secret, error) {
+	if vc.config.traceEnabled {
+		var span *trace.Span
+		vc.ctx, span = trace.StartSpan(vc.ctx, fmt.Sprintf("%s/Create", vc.config.tracePrefix))
+		defer span.End()
+	}
```

Comment:
Thoughts on abstracting this to a helper function that returns the deferred function to call? e.g. something along the lines of this pseudo code:

```go
// trace creates the tracing context and returns the function to call to end the trace.
func (vc *vaultClient) trace(name string) func() {
    if !vc.config.traceEnabled {
        return func() {}
    }

    var span *trace.Span
    vc.ctx, span = trace.StartSpan(
        vc.ctx,
        name,
    )
    return func() { span.End() }
}

func (vc *vaultClient) Create(engine, key, value string) (*api.Secret, error) {

    vc.trace(
        fmt.Sprintf("%s/Create", vc.config.tracePrefix),
    )

    // More code here.
}
```

This would also allow you to mock the `trace` func to provide context helpers to test tracing functionality if we want to.

Reply:
I like this a lot.  I have never loved having to add tracing logic to the top of each function.